### PR TITLE
Prevent navigation when event was cancelled

### PR DESCRIPTION
**Pull request by Tomas2D**
Issue described in #237

### CODE REVIEW

**Comment by Tomas2D** [comment]
ping @omgovich

**Comment by Tomas2D** [comment]
ping @molefrog

File: index.js [hunk 1]
```diff
@@ -108,9 +108,13 @@ export const Link = (props) => {
       )
         return;
 
+      onClick && onClick(event);
+      if (event.defaultPrevented) {
+        return;
+      }
```
Comment (molefrog) [inline index.js:114]: Maybe we save a few bytes by rewriting it as:

```suggestion
      if (!event.defaultPrevented) {
        event.preventDefault();
        navRef.current()
      }
```

?

File: index.js [hunk 2]
```diff
@@ -108,9 +108,13 @@ export const Link = (props) => {
...
+
       event.preventDefault();
       navRef.current();
-      onClick && onClick(event);
```
Comment (molefrog) [inline index.js:113]: I'm trying to think about any possible use case when the order of `onClick` and `navRef()` would matter. It doesn't seem so at the first glance, but maybe @Tomas2D you have some thoughts?

**Review by molefrog — APPROVED** [review]
This looks good, thank you @Tomas2D!

I've left some minor comments and suggestions.

File: index.js (hunk 1 above)
Comment (Tomas2D) [reply index.js:114]: Make sense, why not.

**Review by molefrog — APPROVED** [review]

**Comment by molefrog** [comment]
Thanks for your contribution @Tomas2D!

I'm going to work on releasing the stable 2.8.0 and mention you in the changelog when it's ready.

**Pull request merged**